### PR TITLE
Test class filter.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_control_pane.dart [hunk 1]
```diff
@@ -87,6 +87,9 @@ class SnapshotControlPane extends StatelessWidget {
   }
 }
 
+@visibleForTesting
+const classFilterButtonKey = Key('classFilterButton');
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_control_pane.dart:91]: From what I recall, we typically try and avoid using `Key`s just for testing (cc @kenzieschmoll)

Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_control_pane.dart]: replaced with type

File: packages/devtools_app/test/memory/diff/widgets/class_filter_test.dart [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  WidgetTester tester,
+) async {
+  await _checkDataGolden(from, tester);
+
+  // Open dialog.
+  await tester.tap(find.byType(ClassFilterButton));
+  await _checkFilterGolden(from, tester);
+
+  // Select new filter.
+  await tester.tap(find.byKey(Key(to.toString())));
```
Comment (bkonyi) [inline packages/devtools_app/test/memory/diff/widgets/class_filter_test.dart:134]: We don't need to find this by key, we can just do something like this:

```dart
await tester.tap(find.text(to.toString());
```

Comment (polina-c) [reply packages/devtools_app/test/memory/diff/widgets/class_filter_test.dart:134]: The problem is that tap by text does not work. You need to tab the radio button directly.

**Review by bkonyi — APPROVED** [review]

File: packages/devtools_app/test/memory/diff/widgets/class_filter_test.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/test/memory/diff/widgets/class_filter_test.dart:134]: You can find all the radio keys and then grab the one you want by index. Key's can have unintended side effects and should not be used just for testing where-ever possible. Please change.

**Pull request merged**